helm/freenas-provisioner/templates/serviceaccount.yaml

```diff
@@ -0,0 +1,8 @@
+{{- if .Values.serviceAccount.create -}}
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ include "freenas-provisioner.serviceAccountName" . }}
```

Comment:
When `serviceAccount.create` is true:
  - use a default secretAccountName including the Release.Name to avoid collision

When `serviceAccount.create` is false:
  - do not create a serviceAccount
  - use the default one in the deployment
  - use the default one in the rbac configs